### PR TITLE
get rid of old typealiases

### PR DESCRIPTION
**Pull request by KristofferC**
As the comment in `common.jl` says, these are obsolete now with the terse covariant syntax.

Interestingly, there is apparently a whole package just for these type alises: https://github.com/AP6YC/NumericalTypeAliases.jl/blob/develop/README.md#aliases.

### CODE REVIEW

File: docs/src/counts.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ The package provides functions to count the occurrences of distinct values.
 ```@docs
 counts
 proportions
-addcounts!(r::AbstractArray, x::StatsBase.IntegerArray, levels::StatsBase.IntUnitRange)
+addcounts!(r::AbstractArray, x::StatsBase.AbstractArray{<:Integer}, levels::StatsBase.IntUnitRange)
```
Comment (ararslan) [inline docs/src/counts.md:10]:
```suggestion
addcounts!(r::AbstractArray, x::AbstractArray{<:Integer}, levels::StatsBase.IntUnitRange)
```
I guess `IntUnitRange` could be similarly removed, as it's an alias for `UnitRange{<:Integer}`.

File: docs/src/scalarstats.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ percentile
 iqr
 nquantile
 quantile
-Statistics.median(v::StatsBase.RealVector, w::AbstractWeights{<:Real})
+Statistics.median(v::StatsBase.AbstractVector{<:Real}, w::AbstractWeights{<:Real})
```
Comment (ararslan) [inline docs/src/scalarstats.md:72]:
```suggestion
Statistics.median(v::AbstractVector{<:Real}, w::AbstractWeights{<:Real})
```

**Review by ararslan — COMMENTED** [review]
Thank you for doing this! Though clearly the proper fix would have been to take a dependency on NumericalTypeAliases. :trollface:

**Comment by KristofferC** [comment]
Updated

**Pull request merged**